sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,692 @@
+              new_profile.attribution_content,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(
+              shutdown.attribution_content,
+              'shutdown',
+              DATETIME(shutdown.submission_timestamp)
+            )
+          ),
+          (STRUCT(main.attribution_content, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_content__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (
+            STRUCT(
+              new_profile.attribution_experiment,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(
+              shutdown.attribution_experiment,
+              'shutdown',
+              DATETIME(shutdown.submission_timestamp)
+            )
+          ),
+          (STRUCT(main.attribution_experiment, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_experiment__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (
+            STRUCT(
+              new_profile.attribution_medium,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(shutdown.attribution_medium, 'shutdown', DATETIME(shutdown.submission_timestamp))
+          ),
+          (STRUCT(main.attribution_medium, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_medium__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (
+            STRUCT(
+              new_profile.attribution_source,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(shutdown.attribution_source, 'shutdown', DATETIME(shutdown.submission_timestamp))
+          ),
+          (STRUCT(main.attribution_source, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_source__source_ping,
+      CASE
+        WHEN new_profile.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_new_profile_ping,
+      CASE
+        WHEN shutdown.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_shutdown_ping,
+      CASE
+        WHEN main.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_main_ping
+    ) AS metadata
+  FROM
+    new_profile_ping AS new_profile
+  FULL OUTER JOIN
+    shutdown_ping AS shutdown
+  USING
+    (client_id)
+  FULL OUTER JOIN
+    main_ping AS main
+  USING
+    (client_id)
+  WHERE
+    client_id IS NOT NULL
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.clients_first_seen_v2`
+)
```

Comment:
1. Do we know which fields are initially null then available in a later ping and how much later that field value comes in?  
2. Do we want to consider a hard cap instead of all rows?  Only update 1 week of _previous?

Reply:
Yes, [this query](https://sql.telemetry.mozilla.org/queries/92681/source) is tracking the updates per column of existing records on the current_date. Or in other words how many records from X days ago were updated today. The most updated column is below 0,5% records updated.
On option we talk about with Leif is to set an "update window", where records are updated with later pings only up to one week after the `first_seen_date`. We wanted to track some data for that. Do you think one week window update is a good approach?

Reply:
I do agree with a window approach.

Which column is the most updated and how did you determine the 0,5%?  Shouldn't the % be a list of %?  

E.g for 2023-06-12 **(day0)**
- 0,5% records on 2023-06-11 **(day-1)** were updated
- 0,3% records on 2023-06-10 **(day-2)** were updated 
- ...

I would expect there to be a drop off and that should mark the window size?

Reply:
The query is now updating some of the columns on a 7 day window, not experiments or environment captured-columns.

Reply:
I will say that for 0.5% or even up to 1% improvement, if it changes our overall numbers and we have to wait 1 week to be final I don't think that juice is worth the squeeze. I'd prefer if we just looked as of the day we were running this and not worry about the window. When it looks like numbers are shifting around for users of the data it creates a lot more headache in my experience trying to build trust around them. But this isn't a call for me to make. What do you think @SuYoungHong  and @irrationalagent?

Reply:
Removing the 7-day window based on the numbers (<1% data completeness improvement) and slack discussion with George and Leif.